src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsController.java

```diff
@@ -341,6 +350,43 @@ public ResponseEntity<GetTaskRolePermissionsResponse> getTaskRolePermissions(
             .body(new GetTaskRolePermissionsResponse(taskRolePermissions));
     }
 
+
+    @Operation(description = "Deletes all tasks related to a case.")
+    @ApiResponses({
+        @ApiResponse(responseCode = "201", description = CREATED),
+        @ApiResponse(responseCode = "400", description = BAD_REQUEST),
+        @ApiResponse(responseCode = "403", description = FORBIDDEN),
+        @ApiResponse(responseCode = "500", description = INTERNAL_SERVER_ERROR)
+    })
+    @PostMapping(path = "/delete", consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> deleteTasks(
+            @RequestBody final DeleteTasksRequest deleteTasksRequest,
+            @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthToken) {
+        try {
+            boolean hasAccess = clientAccessControlService.hasExclusiveAccess(serviceAuthToken);
+
+            if (!hasAccess) {
+                return buildErrorResponseEntityAndLogError(HttpStatus.FORBIDDEN.value(),
+                        new GenericForbiddenException(GENERIC_FORBIDDEN_ERROR));
+            }
+
+            verifyCaseId(deleteTasksRequest.getDeleteCaseTasksAction().getCaseRef());
+
+            taskDeletionService.deleteTasksByCaseId(deleteTasksRequest.getDeleteCaseTasksAction().getCaseRef());
+
+            return status(HttpStatus.CREATED.value())
```

Comment:
I think the status code has to be OK than CREATED ? As the business logic underneath is to remove data from the db.

Reply:
The status code is defined in our LLD and agreed with Nigel Barlass - https://tools.hmcts.net/confluence/pages/diffpages.action?originalId=1660290997&pageId=1663966928&__ncforminfo=R7E3CoIvliTg-67RCXoKZ2UoRheYVnEIclYSS1XGCZSRzGnJxzxeY7SO3GWVbFBcVS7U5Tlaus9OTgv_gjBsCWxoMBcw1Q6p3faUIWHmMas=